wandb/sdk/data_types/table.py

```diff
@@ -357,6 +360,19 @@ def handle_resumed_run(
             self._increment_num = increment_num
         self._resume_handled = True
 
+    def ensure_single_run_target(self, run: "LocalRun") -> None:
+        """Handle setting up the run target for incremental table logging.
+
+        This method is called when a table is being logged incrementally to ensure
+        that all increments are logged to the same run.
+        """
+        if self._run_target_for_increments is None:
+            self._run_target_for_increments = run
+        else:
+            assert (
+                self._run_target_for_increments == run
+            ), "Cannot log table incrementally to multiple runs"
```

Comment:
```suggestion
        elif self._run_target_for_increments is not run:
            raise AssertionError("An incremental Table can only be logged to one Run.")
```

Two changes:

1. `is` matches the intent better than `==`. `is` compares identity, but `==` can be overridden.
2. `assert` statements get removed when Python runs in optimized mode. Raising an error explicitly is generally preferred; `assert` statements are best treated like comments.

Reply:
thanks for explaining the nuance here. for the longest time I've only used "is" for null checks.